Add docs (#8)
lukso-network/tools-eip191-signer: README.md

 <p align="center">Helper library to sign a transaction
</p>

> The prefix `\x19LSP6 ExecuteRelayCall:\n` is added to prevent users from signing a transaction while thinking it's a message.

Reviewer: We could do with more context here. Why do we use a prefix here? The prefix is added to what? 

You could have a sentence saying this library is used to sign transactions so they are ready to be passed to the LSP6 Execute Relay Call function. This library will add the LSP6 ExecuteRelayCall prefix to a message and sign it

- [GitHub Repository](https://github.com/lukso-network/tools-lsp6-signer)
- [NPM Package](TODO)

# Install

```bash
npm install @lukso/lsp6-signer.js
```

# Setup

```javascript
import { LSP6Signer } from '@lukso/lsp6-signer.js';

const lsp6Signer = new LSP6Signer();
```

# Usage

## hashMessage

```javascript
lsp6Signer.hashMessage(message);
```

Hashes the given message. The data will be enveloped as follows: `'\x19LSP6 ExecuteRelayCall:\n' + message.length + message` and hashed using keccak256.

### Parameters

> `message` - `String`: A message to hash

Reviewer: We will also need to improve this function in the future by allowing multiple parameters, similar to `web3.utils.soliditySha3`

Reviewer: just noting here

### Returns

> `String`: The hashed message

Reviewer: ```suggestion
`String`: The hashed message with the '\x19LSP6 ExecuteRelayCall:\n' prefix added
```

### Example

```javascript
lsp6Signer.hash('Hello World') >
  '0x90379a0a840499a5e18db4ff7b4d9a56cd009f014a66e3ca86001da3442b7d2c';
```

## sign

```javascript
lsp6Signer.sign(message, signingKey);
```

Signs the message.

### Parameters

`message` - `String`: The message to sign. \
`signingKey` - `String`: The private key to sign with.

> > **_NOTE:_** The value passed as the data parameter will be wrapped as follows: `'\x19LSP6 ExecuteRelayCall:\n' + message.length + message`.

Reviewer: I think this note can go into the main body where you describe what the function does, as it is a crucial component of what the function is for

> > **_NOTE:_** The value passed as the data parameter will be wrapped as follows: `'\x19LSP6 ExecuteRelayCall:\n' + message.length + message`.

Reviewer: ```suggestion
> **_NOTE:_** The value passed as the message parameter will be wrapped as follows: `'\x19LSP6 ExecuteRelayCall:\n' + message.length + message`.
```

### Returns

`Object`: **The Message object**

- `message` - `String`: The given message.
- `messageHash` - `String`: The hash of the given message.
- `r` - `String`: First 32 bytes of the signature.
- `s` - `String`: Next 32 bytes of the signature.
- `v` - `String`: Recovery value + 27.

### Example

```javascript
lsp6Signer.sign(
  'hello',
  'ffeb17b9a6059fec3bbab63d76b060b7380cac7a62ce6621a134531a46458968',
) >
  {
    message: 'hello',
    messageHash:
      '0xd8fd712232755d5f8819c95fdd911dcd51c012b8a66f9710ef0c86ad30925fa6',
    v: '0x1c',
    r: '0x1eab2de0103b8e82650f9706b17cf2adce55a335e7041bad5a94ab49c56a9c12',
    s: '0x662e80a369ffa2a6a77fbeaad1f32653cbd74860c8fbc999b1fc47b8d1cb7d93',
    signature:
      '0x1eab2de0103b8e82650f9706b17cf2adce55a335e7041bad5a94ab49c56a9c12662e80a369ffa2a6a77fbeaad1f32653cbd74860c8fbc999b1fc47b8d1cb7d931c',
  };
```

## recover

```javascript
lsp6Signer.recover(message, signature, isMessagePrefixed);
```

> Recovers the LUKSO address which was used to sign the given data.

Reviewer: The address is not specific to LUKSO. The private key is mathematically linked to the public address so you could use the same private key on ethereum, LUKSO, Binance smart chain etc etc 

Also we want to accommodate  people using these tools on any chain they like, so we can keep these docs generic about the network they are used on

### Parameters

> 1. `message|Message` - `String|Object`: **Either signed message or the Message object with the following values**:

Reviewer: I suggest keeping the name of the parameter the same. We do not need to use the object type here

```suggestion
1. `message` - `String|Object`: **Either signed message or the Message object with the following values**:
```

   - `message` - `String`: The given message.

>    - `messageHash` - `String`: The hash of the given message already prefixed with `'\x19LSP6 ExecuteRelayCall:\n' + message.length + message`.

Reviewer: ```suggestion
   - `messageHash` - `String`: The hash of the given message already prefixed with `'\x19LSP6 ExecuteRelayCall:\n' + message.length`.
```

Or

```suggestion
   - `messageHash` - `String`: The hash of the given message constructed as `'\x19LSP6 ExecuteRelayCall:\n' + message.length + message`.
```

   - `r` - `String`: First 32 bytes of the signature
   - `s` - `String`: Next 32 bytes of the signature
   - `v` - `String`: Recovery value + 27
   - `signature` - `String`: The raw RLP encoded signature.

2. `signature` - `String`: The raw RLP encoded signature.

> 3. `isMessagePrefixed` - `Boolean` (optional, default: false): If the last parameter is true, the given message will NOT automatically be prefixed with `'\x19LSP6 ExecuteRelayCall:\n' + message.length + message`, and assumed to be already prefixed.

Reviewer: I think we don't need to say the last param as this is the description of the specific param we are discussing

```suggestion
3. `isMessagePrefixed` - `Boolean` (optional, default: false): If true, the given message will NOT automatically be prefixed and hashed with `'\x19LSP6 ExecuteRelayCall:\n' + message.length + message`, and assumed to be already prefixed and hashed.
```

### Returns

> `String`: The LUKSO address used to sign this data.

Reviewer: Same as above, we don't need to specify the network as it could be on any

### Example

```javascript

> lsp6signer.recover(

Reviewer: ```suggestion
lsp6Signer.recover(
```
